egs/wsj/s5/steps/libs/nnet3/report/log_parse.py

```diff
@@ -27,6 +28,51 @@ def __init__(self, message = None):
                            "There was an error while trying to parse the logs."
                            " Details : \n{0}\n".format(message))
 
+# This function is used to fill stats_per_component_per_iter table with the 
+# results of regular expression.
+def fill_stats_table_with_regex_result(groups, gate_index, stats_table):
+    iteration = int(groups[0])
+    component_name = groups[1]
+    component_type = groups[2]
+    value_percentiles = groups[3+gate_index*6]
```

Comment:
Do you think you could reduce this verbosity using a for loop and a dictionary ?